diff --git a/lib/base.php b/lib/base.php
index 732fc55be3e..db2c2866b7b 100644
--- a/lib/base.php
+++ b/lib/base.php
@@ -1132,6 +1132,9 @@ class OC {
 		if (OC_User::handleApacheAuth()) {
 			return true;
 		}
+		if (self::tryAppAPILogin($request)) {
+			return true;
+		}
 		if ($userSession->tryTokenLogin($request)) {
 			return true;
 		}
@@ -1169,6 +1172,22 @@ class OC {
 			}
 		}
 	}
+
+	protected static function tryAppAPILogin(OCP\IRequest $request): bool {
+		$appManager = Server::get(OCP\App\IAppManager::class);
+		if (!$request->getHeader('AUTHORIZATION-APP-API')) {
+			return false;
+		}
+		if (!$appManager->isInstalled('app_api')) {
+			return false;
+		}
+		try {
+			$appAPIService = Server::get(OCA\AppAPI\Service\AppAPIService::class);
+			return $appAPIService->validateExAppRequestToNC($request);
+		} catch (\Psr\Container\NotFoundExceptionInterface|\Psr\Container\ContainerExceptionInterface $e) {
+			return false;
+		}
+	}
 }
 
 OC::init();
